src/Documentation/SidebarMenu/SidebarMenu.js

```diff
@@ -19,7 +19,12 @@ const blankStyle = {}
 // Element weight for animations
 const heightMap = {}
 
-// Helper function to calculate element height with all opened children
+/**
+ * Calculate element height with all open children.
+ * @param {Object} params - Parameters needed to calculate height.
+ * @param {Object} params.activePaths - ???.
+ * @param {string} params.path - ???.
+ */
 function calculateHeight({ activePaths, path }) {
```

Comment:
Need help for names and descriptions of these params. I don't think an Object here is really needed also: Shouldn't `activePaths` and `path` be separate params?

Reply:
let's keep that simple comment? even that one is already stretching it in my mind ) Using that heavy jsdoc style is probably to much for helpers ... and probably too much in general )

Reply:
BTW this suggestion is based off an example in https://jsdoc.app/tags-param.html#parameters-with-properties.

Reply:
Yes, you're right. Will revert... But this function needs refactoring for readability :P

Reply:
> But this function needs refactoring for readability.

It might be indeed and it had even a comment :). But let me give you my 2c on how I usually read code:

I would never go into reading a file "as-is", meaning I would never even bother reading all these helpers (why would I care about them unless I need to fix a bug in one of them?). I usually start from some let's call it "entry-point" depending on a task you are trying to solve. And at this stage it's important to have a good project structure that somehow correlates with the real-world concepts so that it's easy to find that entry point. In most cases it means you go and see a component. I don't care about all the stuff around initially. Components are by definition declarative and you can make a lot of sense reading them w/o going too deep. 

Only after that (and having some context) you start jumping into particular functions. Since you have more context it helps to read these helpers in most cases.

So, the point here is that helpers, smaller components sometimes are not meant to be understood w/o understanding the larger context.

Reply:
> I would never go into reading a file "as-is", meaning I would never even bother reading all these helpers... smaller components sometimes are not meant to be understood w/o understanding the larger context

Somewhat agree. Let's try this approach and see how one would find about `function calculateHeight`: See the `SidebarMenu` Component and assume it renders the left-hand site navigation. It uses 1+ `SidebarMenuItem` like this:

https://github.com/iterative/dvc.org/blob/be3fb0ea4a0a6217bd54e01338f83f8f5bfca8d4/src/Documentation/SidebarMenu/SidebarMenu.js#L129-L138

We can then assume the `SidebarMenuItem` PureComponent renders each entry in the nav (sometimes recursively nested other `SidebarMenuItem` instances). SidebarMenuItems contain:

https://github.com/iterative/dvc.org/blob/be3fb0ea4a0a6217bd54e01338f83f8f5bfca8d4/src/Documentation/SidebarMenu/SidebarMenu.js#L62-L72

This is where it gets especially hard to read. I can guess it calculates the height of the collapsable parts (nested SidebarMenuItems). It also explains where the function argument `{ activePaths, path }` comes from.

But at this point it's basically reverse engineering.

Reply:
Anyway, this is probably too deep right now. This change has been reverted.